private/bufpkg/bufimage/util.go

```diff
@@ -424,7 +424,12 @@ func imageToCodeGeneratorRequest(
 		request.Parameter = proto.String(parameter)
 	}
 	for i, imageFile := range imageFiles {
-		request.ProtoFile[i] = imageFile.FileDescriptorProto()
+		// ProtoFile should include runtime-retained options only. So strip source-only options.
```

Comment:
If you implement `ImageFile.FileDescriptorProtoWithoutSourceRetentionOptions()`, then this becomes `if in file_to_generate { use FileDescriptorProto() } else { use FileDescriptorProtoWithoutSourceRetentionOptions() }`

Reply:
Good catch! Silly me for assuming this would be more consistent/coherent. What a terrible API :/

I guess they moved out source-only options for `files_to_generate` just so that existing plugins (not updated for these new fields) that embed descriptors wouldn't need any updates to respect the source-only retention (and inadvertently _not_ embed them). But the fact that `proto_file` behavior is now bimodal feels... dumb.